serving/docs/lmi_new/deployment_guide/configurations.md

```diff
@@ -1,6 +1,189 @@
+You can modify the environment variables as part of your deployment script without having to re-upload configurations to S3.
+This typically leads to a faster iteration loop when modifying and experimenting with configuration values.
+
+While you can mix configurations between `serving.properties` and environment variables, we recommend you choose one and specify all configuration in that format.
+Configurations specified in the `serving.properties` files will override configurations specified in environment variables.
+
+Both configuration mechanisms offer access to the same set of configurations.
+
+If you know which backend you are going to use, you can find a set of starter configurations in the corresponding [user guide](../user_guides).
+We recommend using the quick start configurations as a starting point if you have decided on a particular backend.
+The only change required to the starter configurations is specifying `option.model_id` to point to your model artifacts.
+
+We will now cover the components of a minimal starting configuration. This minimal configuration will look like:
+
+```
+# use standard python engine, or mpi aware python engine
+engine=<Python|MPI>
+# where the model artifacts are stored
+option.model_id=<hf_hub_model_id|s3_uri>
+# which inference library to use
+option.rolling_batch=<auto|vllm|lmi-dist|deepspeed|tensorrtllm>
+# how many gpus to shard the model across with tensor parallelism
+option.tensor_parallelism=<max|number between 1 and number of gpus available>
+```
+
+There are additional configurations that can be specified.
+We will cover the common configurations (across backends) in [LMI Common Configurations](#lmi-common-configurations)
+
+## Model Artifact Configuration
+
+If you are deploying a model hosted on the HuggingFace Hub, you must specify the `option.model_id=<hf_hub_model_id>` configuration.
+When using a model directly from the hub, we recommend you also specify the model revision (commit hash) via `option.revision=<commit hash>`.
+Since model artifacts are downloaded at runtime from the Hub, using a specific revision ensures you are using a model compatible with package versions in the runtime environment.
+Open Source model artifacts on the hub are subject to change at any time, and these changes may cause issues when instantiating the model (the model may require a newer version of transformers than what is available in the container). 
+If a model provides custom model (*modeling.py) and custom tokenizer (*tokenizer.py) files, you need to specify `option.trust_remote_code=true` to load and use the model.
+
+If you are deploying a model hosted in S3, `option.model_id=<s3 uri>` should be the s3 object prefix of the model artifacts.
+Alternatively, you can deploy the model artifacts alongside your `serving.properties` configuration file via the `model.tar.gz` artifact and not specify `option.model_id`.
```

Comment:
for LMI specifically, we do not expect the model to be a `tar.gz`, SageMaker during the runtime will uncompress for us. We can say Sagemaker model object saved into tar.gz. But even today, SM is supporting uncompressed model loading, tar.gz is no longer a requirement

Reply:
Last I tested with `Model` object in sm pysdk, they still hadn't supported uncompressed artifacts. I agree with you, so let me test and validate with pysdk

Reply:
I think I'll just remove this portion for now (tar.gz related content). SM PySDK doesn't support uncompressed model loading even though SM platform does